src/Orleans.Reminders/GrainReminderExtensions.cs

```diff
@@ -0,0 +1,167 @@
+    public static Task UnregisterReminder(this IGrainBase grain, IGrainReminder reminder)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (reminder is null)
+        {
+            throw new ArgumentNullException(nameof(reminder));
+        }
+#endif
+
+        EnsureRuntime(grain);
+
+        return grain.GrainContext.ActivationServices.GetRequiredService<IReminderService>()
+            .UnregisterReminder(reminder);
+    }
+
+    /// <summary>
+    /// Returns a previously registered reminder.
+    /// </summary>
+    /// <param name="grain">The grain instance.</param>
+    /// <param name="reminderName">Reminder to return</param>
+    /// <returns>Promise for Reminder handle.</returns>
+    public static Task<IGrainReminder> GetReminder(this Grain grain, string reminderName) => GetReminder((IGrainBase)grain, reminderName);
+
+    /// <summary>
+    /// Returns a previously registered reminder.
+    /// </summary>
+    /// <param name="grain">A grain.</param>
+    /// <param name="reminderName">Reminder to return</param>
+    /// <returns>Promise for Reminder handle.</returns>
+    public static Task<IGrainReminder> GetReminder(this IGrainBase grain, string reminderName)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (grain is null)
+        {
+            throw new ArgumentNullException(nameof(grain));
+        }
+#endif
+
+        if (string.IsNullOrWhiteSpace(reminderName))
+        {
+            throw new ArgumentNullException(nameof(reminderName));
+        }
+
+        EnsureRuntime(grain);
+
+        return grain.GrainContext.ActivationServices.GetRequiredService<IReminderService>()
+            .GetReminder(grain.GrainContext.GrainReference, reminderName);
+    }
+
+    /// <summary>
+    /// Returns a list of all reminders registered by the grain.
+    /// </summary>
+    /// <returns>Promise for list of Reminders registered for this grain.</returns>
+    public static Task<List<IGrainReminder>> GetReminders(this Grain grain) => GetReminders((IGrainBase)grain);
+
+    /// <summary>
+    /// Returns a list of all reminders registered by the grain.
+    /// </summary>
+    /// <returns>Promise for list of Reminders registered for this grain.</returns>
+    public static Task<List<IGrainReminder>> GetReminders(this IGrainBase grain)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (grain is null)
+        {
+            throw new ArgumentNullException(nameof(grain));
+        }
+#endif
+        EnsureRuntime(grain);
+
+        return grain.GrainContext.ActivationServices.GetRequiredService<IReminderService>().GetReminders(grain.GrainContext.GrainReference);
+    }
+
+    private static void EnsureRuntime(IGrainBase grain)
+    {
+        if (grain.GrainContext.ActivationServices.GetRequiredService<IGrainRuntime>() == null)
```

Comment:
This condition will never be null. I also think this is the wrong way to check if we're running in the grain's context.